### PR TITLE
update min version of azure-mgmt-core in shared reqs

### PR DESCRIPTION
**Pull request by kristapratico**
https://dev.azure.com/azure-sdk/public/_build/results?buildId=1414764&view=results

This PR: https://github.com/Azure/azure-sdk-for-python/pull/23332
 bumped the azure-mgmt-core version for most, but not all mgmt packages. The shared reqs needs to be updated to reflect that.

### CODE REVIEW

File: shared_requirements.txt [hunk 1]
```diff
@@ -303,3 +303,24 @@ opentelemetry-sdk<2.0.0,>=1.5.0,!=1.10a0
 #override azure-mgmt-sqlvirtualmachine azure-mgmt-core>=1.3.0,<2.0.0
 #override azure-mgmt-rdbms msrest>=0.6.21
 #override azure-mgmt-rdbms azure-mgmt-core>=1.3.0,<2.0.0
```
Comment (mccoyp) [inline shared_requirements.txt:305]: We could remove a lot of these `azure-mgmt-core` overrides with this change; up to you!

**Review by mccoyp — APPROVED** [review]
Thank you for getting to this before me 🙂

File: shared_requirements.txt (hunk 1 above)
Comment (kristapratico) [reply shared_requirements.txt:305]: good point, let me do that!

Comment (kristapratico) [reply shared_requirements.txt]: Analyze dependencies step is passing CI, merging this

**Comment by xiangyan99** [comment]
I created same PR. https://github.com/Azure/azure-sdk-for-python/pull/23376:)

**Comment by kristapratico** [comment]
> I created same PR. #23376:)

Oh no! I missed that. We're all so on top of it 😄

**Pull request merged**